ratings/ratings.py

```diff
@@ -0,0 +1,298 @@
+    return rowid_for(db, bucket, name_or_num)
+
+def model_num_for(model_id):
+    try:
+        db = sqlite3.connect("ratings.db")
+        return db.execute("select model_num from models where id = ?", (model_id,)).fetchone()[0]
+    except:
+        print("No model found for id: {}".format(model_id))
+        raise
+        return None
+
+def rowid_for(db, bucket,name):
+    try:
+        return db.execute("select id from models where bucket = ? and model_name = ?", [bucket, name]).fetchone()[0]
+    except:
+        print("No row found for bucket: {} name: {}".format(bucket, name))
+        return None
+
+
+def import_files(files, bucket=None):
+    if bucket is None:
+        bucket = fsdb.models_dir()
+
+    print("Importing for bucket:", bucket)
+    db = sqlite3.connect("ratings.db")
+    new_games = 0
+    with db:
+        c = db.cursor()
+        for _file in tqdm(files):
+            match = re.match(EVAL_REGEX, os.path.basename(_file))
+            if not match:
+                print("Bad file: ", _file)
+                continue
+            timestamp = match.groups(1)[0]
+            with open(_file) as f:
+                text = f.read()
+            pw = re.search(PW_REGEX, text)
+            pb = re.search(PB_REGEX, text)
+            result = re.search(RESULT_REGEX, text)
+            if not pw or not pb or not result:
+                print("Fields not found: ", _file)
+
+            pw = pw.group(1)
+            pb = pb.group(1)
+            result = result.group(1)
+
+            m_num_w = re.match(MODEL_REGEX, pw).group(1)
+            m_num_b = re.match(MODEL_REGEX, pb).group(1)
+            if m_num_w == '000588' or m_num_b == '000588':
+                continue
+
+
+
+            try:
+                # create models or ignore.
+                maybe_insert_model(db, bucket, pb, m_num_b)
+                maybe_insert_model(db, bucket, pw, m_num_w)
+
+                b_id = rowid_for(db, bucket, pb)
+                w_id = rowid_for(db, bucket, pw)
+
+                # insert into games or bail
+                game_id = None
+                try:
+                    with db:
+                        c = db.cursor()
+                        c.execute(
+                        """ insert into games(timestamp, filename, b_id, w_id, black_won, result)
+                                        values(?, ?, ?, ?, ?, ?)
+                        """, [timestamp, os.path.relpath(_file), b_id, w_id, result.lower().startswith('b'), result])
+                        game_id = c.lastrowid
+                except sqlite3.IntegrityError:
+                    #print("Duplicate game: {}".format(_file))
+                    continue
+
+                if game_id is None:
+                    print("Somehow, game_id was None")
+
+            # update wins/game counts on model, and wins table.
+                c.execute("update models set num_games = num_games + 1 where id in (?, ?)", [b_id, w_id])
```

Comment:
these can be done after all inserts with

update models set num_games = (select sum(games) from games where b_id = id or w_id = id );